presto-main/src/main/java/com/facebook/presto/type/FloatOperators.java

```diff
@@ -70,4 +78,61 @@ public static long negate(@SqlType(StandardTypes.FLOAT) long value)
     {
         return Float.floatToRawIntBits(-Float.intBitsToFloat((int) value));
     }
+
+    @ScalarOperator(EQUAL)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean equal(@SqlType(StandardTypes.FLOAT) long left, @SqlType(StandardTypes.FLOAT) long right)
+    {
+        return Float.intBitsToFloat((int) left) == Float.intBitsToFloat((int) right);
```

Comment:
Is there a reason simply comparing the longs wouldn't work? Are there floats that compare equal when their bits aren't?